meson.build

```diff
@@ -198,6 +198,27 @@ if get_option('experimental')
 	config.set('ENABLE_EXPERIMENTAL', true)
 endif
 
+if cc.has_function(
+	'getrandom',
+	prefix: '#include <stddef.h>\n#include <sys/random.h>',
+	args: [ '-D_GNU_SOURCE' ]
+)
+	config.set('HAVE_GETRANDOM', true)
+endif
+
+if cc.has_function(
+	'arc4random_buf',
+	prefix: '#include <stdlib.h>'
+)
+	config.set('HAVE_ARC4RANDOM', true)
+endif
+
+if (config.has('HAVE_CRYPTO') and
```

Comment:
Almost. This check should take place where `HAVE_CRYPTO` is set; i.e. inside the `if nettle.found() and hogweed.found() and gmp.found()` block.

It would also be nicer to not use `config.has` here. Declare a meson variable instead that's named something like `have_random`. It would be duplicate state, but using the config file data like this feels weird.

Reply:
Updated...